llvm/test/Transforms/InstCombine/erf.ll

```diff
@@ -0,0 +1,185 @@
+; NOTE: Assertions have been autogenerated by utils/update_test_checks.py UTC_ARGS: --version 5
+; RUN: opt < %s -passes=instcombine -S | FileCheck %s
+
+define float @erff_const() {
+; CHECK-LABEL: define float @erff_const() {
+; CHECK-NEXT:    [[R:%.*]] = call float @erff(float 5.000000e-01)
+; CHECK-NEXT:    ret float 0x3FE0A7EF60000000
+;
+  %r = call float @erff(float 5.000000e-01)
+  ret float %r
+}
+
+define double @erf_const() {
+; CHECK-LABEL: define double @erf_const() {
+; CHECK-NEXT:    [[R:%.*]] = call double @erf(double -5.000000e-01)
+; CHECK-NEXT:    ret double 0xBFE0A7EF5C18EDD2
+;
+  %r = call double @erf(double -5.000000e-01)
+  ret double %r
+}
+
+define float @erff_zero() {
+; CHECK-LABEL: define float @erff_zero() {
+; CHECK-NEXT:    [[R:%.*]] = call float @erff(float 0.000000e+00)
+; CHECK-NEXT:    ret float 0.000000e+00
+;
+  %r = call float @erff(float 0.000000e+00)
+  ret float %r
+}
+
+define double @erf_zero() {
+; CHECK-LABEL: define double @erf_zero() {
+; CHECK-NEXT:    [[R:%.*]] = call double @erf(double 0.000000e+00)
+; CHECK-NEXT:    ret double 0.000000e+00
+;
+  %r = call double @erf(double 0.000000e+00)
```

Comment:
None of these cases folded?

Reply:
The call wasn't deleted though.

Plus the linux docs suggest it doesn't ever set errno

Reply:
> Other libcall test case exhibits similar behavior, but it does not seem to affect the final fold result. I believe the unused instruction will be deleted anyway?

Then those are also broken. This call needs to be immediately deleted

Reply:
Hm, it does seem these are deleted later. I would still expect them to be immediately killed, but this seems to be a more widespread issue